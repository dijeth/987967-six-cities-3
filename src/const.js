const OfferType = {
  APARTMENT: `Apartment`,
  PRIVATE_ROOM: `Private room`
};

const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export {OfferType, CITIES};
