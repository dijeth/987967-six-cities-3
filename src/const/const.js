"use strict";
exports.__esModule = true;
exports.OfferType = {
    APARTMENT: "Apartment",
    PRIVATE_ROOM: "Private room"
};
exports.CITIES = [
    "Paris",
    "Cologne",
    "Brussels",
    "Amsterdam",
    "Hamburg",
    "Dusseldorf"
];
exports.InsideFeature = {
    WI_FI: "Wi-Fi",
    WASHING_MACHINE: "Washing machine",
    TOWELS: "Towels",
    HEATING: "Heating",
    COFFEE_MACHINE: "Coffee machine",
    BABY_SEAT: "Baby seat",
    KITCHEN: "Kitchen",
    DISHWASHER: "Dishwasher",
    CABEL_TV: "Cabel TV",
    FRIDGE: "Fridge"
};
exports.ScreenType = {
    MAIN: "MAIN",
    PROPERTY: "PROPERTY"
};
exports.BREAK_STRING = "\n";
exports.MONTHS = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
exports.MAX_CITY_COUNT = 6;
var SortType;
(function (SortType) {
    SortType["POPULAR"] = "Popular";
    SortType["PRICE_LOW_TO_HIGH"] = "Price: low to high";
    SortType["PRICE_HIGH_TO_LOW"] = "Price: high to low";
    SortType["TOP_RATED_FIRST"] = "Top rated first";
})(SortType = exports.SortType || (exports.SortType = {}));
exports.SORT_LIST = [
    SortType.POPULAR,
    SortType.PRICE_LOW_TO_HIGH,
    SortType.PRICE_HIGH_TO_LOW,
    SortType.TOP_RATED_FIRST
];
exports.AuthorizationStatus = {
    NO_AUTH: "NO_AUTH",
    AUTH: "AUTH"
};
exports.AppRoute = {
    getLogin: function () { return "/login"; },
    getOffer: function (id) { return "/offer/" + id; },
    getFavorites: function () { return "/favorites"; },
    getRoot: function () { return "/"; }
};
exports.ServerRoute = {
    getLogin: function () { return "/login"; },
    getFavorites: function () { return "/favorite"; },
    postFavorites: function (id, status) { return "/favorite/" + id + "/" + status; },
    getNearby: function (id) { return "/hotels/" + id + "/nearby"; },
    getHotels: function () { return "/hotels"; },
    getComments: function (id) { return "/comments/" + id; }
};
exports.MAX_IMAGE_COUNT = 6;
exports.ReviewLength = {
    MIN: 50,
    MAX: 300
};
exports.ServerError = {
    UNAUTHORIZED: 401,
    BAD_REQUEST: 400,
    NOT_FOUND: 404
};
var PlaceCardType;
(function (PlaceCardType) {
    PlaceCardType["DEFAULT"] = "DEFAULT";
    PlaceCardType["FAVORITE"] = "FAVORITE";
    PlaceCardType["NEARBY"] = "NEARBY";
})(PlaceCardType = exports.PlaceCardType || (exports.PlaceCardType = {}));
exports.EMPTY_REVIEW = {
    text: "",
    rating: 0,
    offerID: null
};
exports.BASE_URL = "https://htmlacademy-react-3.appspot.com/six-cities";
exports.TIME_OUT = 5000;
