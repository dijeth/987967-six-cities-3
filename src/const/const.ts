export const OfferType = {
  APARTMENT: `Apartment`,
  PRIVATE_ROOM: `Private room`
};

export const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const InsideFeature = {
  WI_FI: `Wi-Fi`,
  WASHING_MACHINE: `Washing machine`,
  TOWELS: `Towels`,
  HEATING: `Heating`,
  COFFEE_MACHINE: `Coffee machine`,
  BABY_SEAT: `Baby seat`,
  KITCHEN: `Kitchen`,
  DISHWASHER: `Dishwasher`,
  CABEL_TV: `Cabel TV`,
  FRIDGE: `Fridge`
};

export const ScreenType = {
  MAIN: `MAIN`,
  PROPERTY: `PROPERTY`
};

export const BREAK_STRING = `\n`;

export const MONTHS = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];

export const MAX_CITY_COUNT = 6;

export enum SortType {
  POPULAR = `Popular`,
  PRICE_LOW_TO_HIGH = `Price: low to high`,
  PRICE_HIGH_TO_LOW = `Price: high to low`,
  TOP_RATED_FIRST = `Top rated first`,
};

export const SORT_LIST = [
  SortType.POPULAR,
  SortType.PRICE_LOW_TO_HIGH,
  SortType.PRICE_HIGH_TO_LOW,
  SortType.TOP_RATED_FIRST
];

export const AuthorizationStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`,
};

export const AppRoute = {
  getLogin: () => `/login`,
  getOffer: (id) => `/offer/${id}`,
  getFavorites: () => `/favorites`,
  getRoot: () => `/`,
};

export const ServerRoute = {
  getLogin: () => `/login`,
  getFavorites: () => `/favorite`,
  postFavorites: (id, status) => `/favorite/${id}/${status}`,
  getNearby: (id) => `/hotels/${id}/nearby`,
  getHotels: () => `/hotels`,
  getComments: (id) => `/comments/${id}`,
};

export const MAX_IMAGE_COUNT = 6;

export const ReviewLength = {
  MIN: 50,
  MAX: 300
};

export const ServerError = {
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
  NOT_FOUND: 404
};

export enum PlaceCardType {
  DEFAULT = `DEFAULT`,
  FAVORITE = `FAVORITE`,
  NEARBY = `NEARBY`,
};

export const EMPTY_REVIEW = {
  text: ``,
  rating: 0,
  offerID: null,
};

export const BASE_URL = `https://htmlacademy-react-3.appspot.com/six-cities`;
export const TIME_OUT = 5000;
