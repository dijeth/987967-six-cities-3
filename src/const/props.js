import PropTypes from 'prop-types';

export const offerPropType = PropTypes.shape({
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  pictures: PropTypes.arrayOf(PropTypes.string).isRequired,
  cost: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
  isPremium: PropTypes.bool.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  city: PropTypes.string.isRequired,
  coord: PropTypes.arrayOf(PropTypes.number).isRequired
});

export const cityPropType = PropTypes.shape({
  name: PropTypes.string.isRequired,
  zoom: PropTypes.number.isRequired,
  centerCoord: PropTypes.arrayOf(PropTypes.number)
});
