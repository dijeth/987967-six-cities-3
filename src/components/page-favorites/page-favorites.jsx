import React from 'react';
import PropTypes from 'prop-types';
import OffersFavorite from '../offers-favorite/offers-favorite.jsx';
import Header from '../header/header';
import {connect} from 'react-redux';
import {getAuthorizationStatus} from '../../reducers/user/selectors.js';
import {getFavorites} from '../../reducers/data/selectors.js';
import {Link} from 'react-router-dom';
import {AppRoute} from '../../const/const';

const PageFavorites = ({isAuth, favoriteData}) => {
  const {offers: favoriteOffers, cities} = favoriteData;
  const classList = `page__main page__main--favorites ${favoriteOffers.length === 0 ? `page__main--favorites-empty` : ``}`;

  return (
    <div className="page">
      <Header isActiveLogo={false}/>

      <main className={classList}>
        <div className="page__favorites-container container">
          {cities.length === 0 ? (
            <section className="favorites favorites--empty">
              <h1 className="visually-hidden">Favorites (empty)</h1>
              <div className="favorites__status-wrapper">
                <b className="favorites__status">Nothing yet saved.</b>
                <p className="favorites__status-description">Save properties to narrow down search or plan yor future trips.</p>
              </div>
            </section>) : (
            <section className="favorites">
              <h1 className="favorites__title">Saved listing</h1>
              <OffersFavorite offers={favoriteOffers} cities={cities} isAuth={isAuth} />
            </section>)}
        </div>
      </main>
      <footer className="footer container">
        <Link to={AppRoute.getRoot()} className="footer__logo-link" href="main.html">
          <img className="footer__logo" src="img/logo.svg" alt="6 cities logo" width="64" height="33" />
        </Link>
      </footer>
    </div>);
};

PageFavorites.propTypes = {
  isAuth: PropTypes.bool,
  favoriteData: PropTypes.shape({
    offers: PropTypes.object,
    cities: PropTypes.arrayOf(PropTypes.string),
  }),
};

const mapStateToProps = (state) => ({
  isAuth: getAuthorizationStatus(state),
  favoriteData: getFavorites(state)
});

export {PageFavorites};
export default connect(mapStateToProps)(PageFavorites);
