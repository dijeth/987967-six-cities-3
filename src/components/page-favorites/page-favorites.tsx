import * as React from 'react';
import OffersFavorite from '../offers-favorite/offers-favorite';
import Header from '../header/header';
import {connect} from 'react-redux';
import {getAuthorizationStatus} from '../../reducers/user/selectors.js';
import {getFavorites} from '../../reducers/data/selectors.js';
import {Link} from 'react-router-dom';
import {AppRoute} from '../../const/const';
import {OfferMini} from '../../interfaces';

type Props = {
  isAuth: boolean;
  favoriteData: {
    offers: Array<OfferMini>;
    cities: Array<string>;
  };
};

const PageFavorites: React.FC<Props> = ({isAuth, favoriteData}) => {
  const {offers: favoriteOffers, cities} = favoriteData;
  const classList = `page__main page__main--favorites ${favoriteOffers.length === 0 ? `page__main--favorites-empty` : ``}`;

  return (
    <div className="page">
      <Header isActiveLogo={false}/>

      <main className={classList}>
        <div className="page__favorites-container container">
          {cities.length === 0 ? (
            <section className="favorites favorites--empty">
              <h1 className="visually-hidden">Favorites (empty)</h1>
              <div className="favorites__status-wrapper">
                <b className="favorites__status">Nothing yet saved.</b>
                <p className="favorites__status-description">Save properties to narrow down search or plan yor future trips.</p>
              </div>
            </section>) : (
            <section className="favorites">
              <h1 className="favorites__title">Saved listing</h1>
              <OffersFavorite offers={favoriteOffers} cities={cities} isAuth={isAuth} />
            </section>)}
        </div>
      </main>
      <footer className="footer container">
        <Link to={AppRoute.getRoot()} className="footer__logo-link" href="main.html">
          <img className="footer__logo" src="img/logo.svg" alt="6 cities logo" width="64" height="33" />
        </Link>
      </footer>
    </div>);
};

const mapStateToProps = (state) => ({
  isAuth: getAuthorizationStatus(state),
  favoriteData: getFavorites(state)
});

export {PageFavorites};
export default connect(mapStateToProps)(PageFavorites);
