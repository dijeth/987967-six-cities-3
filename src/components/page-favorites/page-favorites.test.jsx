import React from 'react';
import Enzyme from 'enzyme';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
import {PageFavorites} from './page-favorites.jsx';
import NameSpace from '../../reducers/name-space.js';
import {AuthorizationStatus} from '../../const/const';
import {BrowserRouter} from 'react-router-dom';

Enzyme.configure({
  adapter: new Adapter()
});

const mockStore = configureStore([]);

const store = mockStore({
  [NameSpace.APP]: {
    activeOffer: null,
  },

  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.AUTH
  }
});

const favorites = {
  offers: {
    [`Dusseldorf`]: [
      {
        id: `1`,
        city: `Dusseldorf`,
        pictures: [
          `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg`,
          `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg`,
          `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`
        ],
        title: `Nice, cozy, warm big bed apartment`,
        isFavorite: true,
        isPremium: true,
        rating: 3.5,
        type: `house`,
        cost: 868,
        isSuperUser: true,
        userID: `25`,
        coord: [51.237402, 6.797314],
        zoom: 16
      }
    ],
    [`Brussels`]: [
      {
        id: `2`,
        city: `Brussels`,
        pictures: [
          `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg`,
          `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg`,
          `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg`,
          `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg`
        ],
        title: `The Pondhouse - A Magical Place`,
        isFavorite: true,
        isPremium: false,
        rating: 4.5,
        type: `room`,
        cost: 270,
        isSuperUser: true,
        userID: `25`,
        coord: [50.869557, 4.332697],
        zoom: 16
      }
    ]
  },

  cities: [`Brussels`, `Dusseldorf`]
};

it(`<PageFavorites /> should be rendered correctly`, () => {
  const tree = Enzyme.mount(
      <Provider store={store}>
        <BrowserRouter>
          <PageFavorites isAuth={true} favoriteData={favorites} />
        </BrowserRouter>
      </Provider>
  );
  expect(tree.getDOMNode()).toMatchSnapshot();
});
