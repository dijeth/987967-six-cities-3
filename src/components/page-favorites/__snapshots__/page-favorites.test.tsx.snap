// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageFavorites /> should be rendered correctly 1`] = `
<div
  class="page"
>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link"
            href="/"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
          >
            <li
              class="header__nav-item user"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/favorites"
              >
                <div
                  class="header__avatar-wrapper  user__avatar-wrapper"
                />
                <span
                  class="header__user-name user__name"
                />
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    class="page__main page__main--favorites "
  >
    <div
      class="page__favorites-container container"
    >
      <section
        class="favorites"
      >
        <h1
          class="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          class="favorites__list"
        >
          <li
            class="favorites__locations-items"
          >
            <div
              class="favorites__locations locations locations--current"
            >
              <div
                class="locations__item"
              >
                <a
                  class="locations__item-link"
                  href="/"
                >
                  <span>
                    Brussels
                  </span>
                </a>
              </div>
            </div>
            <div
              class="favorites__places"
            >
              <article
                class="favorites__card place-card"
              >
                <div
                  class="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      class="place-card__image"
                      height="110"
                      src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  class="favorites__card-info place-card__info"
                >
                  <div
                    class="place-card__price-wrapper"
                  >
                    <div
                      class="place-card__price"
                    >
                      <b
                        class="place-card__price-value"
                      >
                        €
                        270
                      </b>
                      <span
                        class="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      class="place-card__bookmark-button place-card__bookmark-button--active button"
                      type="button"
                    >
                      <svg
                        class="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlink:href="#icon-bookmark"
                        />
                      </svg>
                      <span
                        class="visually-hidden"
                      >
                        From bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    class="place-card__rating rating"
                  >
                    <div
                      class="place-card__stars rating__stars"
                    >
                      <span
                        style="width: 100%;"
                      />
                      <span
                        class="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    class="place-card__name"
                  >
                    <a
                      href="/offer/2"
                    >
                      The Pondhouse - A Magical Place
                    </a>
                  </h2>
                  <p
                    class="place-card__type"
                  >
                    room
                  </p>
                </div>
              </article>
            </div>
          </li>
          <li
            class="favorites__locations-items"
          >
            <div
              class="favorites__locations locations locations--current"
            >
              <div
                class="locations__item"
              >
                <a
                  class="locations__item-link"
                  href="/"
                >
                  <span>
                    Dusseldorf
                  </span>
                </a>
              </div>
            </div>
            <div
              class="favorites__places"
            >
              <article
                class="favorites__card place-card"
              >
                <div
                  class="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  class="favorites__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="#"
                  >
                    <img
                      alt="Place image"
                      class="place-card__image"
                      height="110"
                      src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg"
                      width="150"
                    />
                  </a>
                </div>
                <div
                  class="favorites__card-info place-card__info"
                >
                  <div
                    class="place-card__price-wrapper"
                  >
                    <div
                      class="place-card__price"
                    >
                      <b
                        class="place-card__price-value"
                      >
                        €
                        868
                      </b>
                      <span
                        class="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      class="place-card__bookmark-button place-card__bookmark-button--active button"
                      type="button"
                    >
                      <svg
                        class="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlink:href="#icon-bookmark"
                        />
                      </svg>
                      <span
                        class="visually-hidden"
                      >
                        From bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    class="place-card__rating rating"
                  >
                    <div
                      class="place-card__stars rating__stars"
                    >
                      <span
                        style="width: 80%;"
                      />
                      <span
                        class="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    class="place-card__name"
                  >
                    <a
                      href="/offer/1"
                    >
                      Nice, cozy, warm big bed apartment
                    </a>
                  </h2>
                  <p
                    class="place-card__type"
                  >
                    house
                  </p>
                </div>
              </article>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    class="footer container"
  >
    <a
      class="footer__logo-link"
      href="/"
    >
      <img
        alt="6 cities logo"
        class="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
